### PR TITLE
Application services

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

This converts the top level application objects into services and uses service injection for booting and executing them.

#### Testing Instructions

CLI & Web apps should still work correctly.

### CODE REVIEW

File: cli/tracker.php [hunk 1]
```diff
@@ -35,9 +33,83 @@
...
+					'Exception of type %1$s thrown while booting the application',
+					get_class($e)
+				),
+				['exception' => $e]
+			);
+		}
+		catch (\Exception $nestedException)
+		{
+			// Do nothing, we tried our best
+		}
```
Comment (elkuku) [inline cli/tracker.php]: Not sure if we should try to do "something" here, maybe just echoing the message of the `$nestedException`?

File: www/index.php [hunk 2]
```diff
@@ -23,21 +23,99 @@
...
+				sprintf(
+					'Exception of type %1$s thrown while booting the application',
+					get_class($e)
+				),
+				['exception' => $e]
+			);
+		}
+		catch (\Exception $nestedException)
+		{
+			// Do nothing, we tried our best
```
Comment (elkuku) [inline www/index.php]: Do "something"?

File: www/index.php [hunk 3]
```diff
@@ -23,21 +23,99 @@
...
+				'Exception of type %1$s thrown while executing the application',
+				get_class($e)
+			),
+			['exception' => $e]
+		);
+	}
+	catch (\Exception $nestedException)
+	{
+		// Do nothing, we tried our best
+	}
```
Comment (elkuku) [inline www/index.php]: Do "something"?

**Comment by elkuku** [comment]
There seems to be a lot of useful stuff in here. I'm sorry I really didn't got a chance to get my head around this, until now. Would you mind bringing this to a mergeable state again?

File: cli/tracker.php (hunk 1 above)
Comment (mbabker) [inline cli/tracker.php]: IMO it's not a big concern here as the nested exception would be an error writing to the log file.  The actual error causing the script to fail is spit out below.

File: www/index.php (hunk 2 above)
Comment (mbabker) [inline www/index.php]: Same argument as in the CLI front controller.  Can't say I'm too concerned about an error writing to the log file; the pertinent error message is displayed on screen.

**Comment by mbabker** [comment]
master branch is merged

**Pull request merged**